packages/foam-vscode/src/features/hover-provider.ts

```diff
@@ -0,0 +1,83 @@
+import * as vscode from 'vscode';
+import { Foam, FoamWorkspace, ResourceParser, URI, Range } from 'foam-core';
+import { FoamFeature } from '../types';
+import { getNoteTooltip, mdDocSelector, isSome } from '../utils';
+import { toVsCodeRange } from '../utils/vsc-utils';
+import { ResourceLink } from 'foam-core';
+import {
+  ConfigurationMonitor,
+  monitorFoamVsCodeConfig,
+} from '../services/config';
+
+export const CONFIG_KEY = 'links.hover.enable';
+
+const feature: FoamFeature = {
+  activate: async (
+    context: vscode.ExtensionContext,
+    foamPromise: Promise<Foam>
+  ) => {
+    const isHoverEnabled: ConfigurationMonitor<boolean> = monitorFoamVsCodeConfig(
+      CONFIG_KEY
+    );
+
+    const foam = await foamPromise;
+
+    context.subscriptions.push(
+      isHoverEnabled,
+      vscode.languages.registerHoverProvider(
+        mdDocSelector,
+        new HoverProvider(isHoverEnabled, foam.workspace, foam.services.parser)
+      )
+    );
+  },
+};
+
+export class HoverProvider implements vscode.HoverProvider {
+  constructor(
+    private isHoverEnabled: () => boolean,
+    private workspace: FoamWorkspace,
+    private parser: ResourceParser
+  ) {}
+
+  async provideHover(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    token: vscode.CancellationToken
+  ): Promise<vscode.Hover> {
+    if (!this.isHoverEnabled()) {
+      return;
+    }
+
+    const startResource = this.parser.parse(document.uri, document.getText());
+
+    const targetLink: ResourceLink | undefined = startResource.links.find(
+      link =>
+        link.type === 'wikilink' &&
```

Comment:
why limit this feature to wikilinks? can't we add a hover for `[direct links](./path.md)` too? Foam supports them

Reply:
tested locally, even just removing the line will make it work. we should add a test case though

Reply:
At first, I thought the semantics of direct links were significantly different from wikilinks.

Now it works for both https://github.com/foambubble/foam/pull/728/commits/beee772f9c9e8e026959c06f13f2d29fe275b899